### PR TITLE
Reworked the EventHandle class to support different internal types

### PR DESCRIPTION
**Pull request by StevenFrost**
This is primarily to support some `SimConnectClient` work where we have handles which require a different internal type to the existing `EventHandle`. There are also some slight API changes to allow use of handles in different ways (eg. exposing the internal ID so we can pass that to SimConnect).

### CODE REVIEW

File: Source/Utils/Inc/Utils/Handle/Handle.h [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	}
+
+	void Reset()
+	{
+		Id = InvalidId;
+	}
+
+private:
+
+	static TId	InvalidId;
```
Comment (ollyau) [inline Source/Utils/Inc/Utils/Handle/Handle.h]: No longer const?

File: Source/Utils/Inc/Utils/Handle/Handle.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	static TId	NextId;
+
+	TId			Id;
+};
+
+// -----------------------------------------------------------------------------
+
+template< class TId > TId Handle< TId >::InvalidId = 0;
+template< class TId > TId Handle< TId >::NextId = 1;
```
Comment (ollyau) [inline Source/Utils/Inc/Utils/Handle/Handle.h]: If `NextId` starts at 1, and `TId` is signed (e.g. in the above `typedef Handle< int32_t > EventHandle;`), do we effectively lose half the possible values?

File: Source/Utils/Inc/Utils/Handle/Handle.h [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+{
+public:
+
+	static_assert( std::is_integral< TId >(), "TId must be an integral type" );
+
+	static Handle Make()
+	{
+		if ( NextId == std::numeric_limits< TId >::max() )
+		{
+			return Handle();
```
Comment (ollyau) [inline Source/Utils/Inc/Utils/Handle/Handle.h:46]: Do we want to add any logging here?

**Review by ollyau — COMMENTED** [review]
Just a couple questions!

File: Source/Utils/Inc/Utils/Handle/Handle.h (hunk 1 above)
Comment (StevenFrost) [reply Source/Utils/Inc/Utils/Handle/Handle.h]: Good spot, a mistake when I was refactoring! Will no longer be required in the next update to this PR.

File: Source/Utils/Inc/Utils/Handle/Handle.h (hunk 2 above)
Comment (StevenFrost) [reply Source/Utils/Inc/Utils/Handle/Handle.h]: Yeah, going to rework this a little so we use the full range of the type & allow customisation if needed by specific types.

**Review by ollyau — APPROVED** [review]
The additional changes look good! :)

**Review by ollyau — APPROVED** [review]
Approved again!  Thanks for making it more consistent with the other usage. :P

**Pull request merged**